src/TriggerActivityMakerChannelAdjacency.cpp

```diff
@@ -34,10 +35,21 @@ TriggerActivityMakerChannelAdjacency::operator()(const TriggerPrimitive& input_t
   // The first time operator() is called, reset the window object.
   if (m_current_window.is_empty()) {
     m_current_window.reset(input_tp);
-    m_primitive_count++;
     return;
   }
-  
+
+  if (m_first_tp){ 
+  using namespace std::chrono;
+  // If this is the first TP of thm_data_vs_system_time_oute run, calculate the initial offset:
+    m_initial_offset = (duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count()) - (input_tp.time_start*(16*1e-6));
+    m_first_tp = false;
+  }
+
+  // Update OpMon Variable(s)
+  uint64_t system_time = duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count();
+  uint64_t data_time = input_tp.time_start*(16*1e-6);                              // Convert 62.5 MHz ticks to ms
```

Comment:
Same as in the other PR with the magic numbers.
Also, isn't `16*1e-6` the same as `16e-6`? The extra `*` will take extra cycle from a processor (which isn't much, so doesn't matter 🤷🏻 )